x-pack/plugins/rule_registry/common/types.ts

```diff
@@ -302,3 +305,21 @@ export interface ClusterPutComponentTemplateBody {
     mappings: estypes.MappingTypeMapping;
   };
 }
+
+export interface BrowserField {
+  aggregatable: boolean;
+  category: string;
+  description?: string | null;
+  example?: string | number | null;
+  fields: Readonly<Record<string, Partial<BrowserField>>>;
+  format?: string;
+  indexes: string[];
+  name: string;
+  searchable: boolean;
+  type: string;
+  subType?: IFieldSubType;
+  readFromDocValues: boolean;
+  runtimeField?: RuntimeField;
+}
+
+export type BrowserFields = Record<string, Partial<BrowserField>>;
```

Comment:
Wondering why we are using `Partial` with the `BrowserField` value, shouldn't it have the required fields at least? Using this type an empty object is a valid BrowserField.

Reply:
yes, I am okay with that 👍 
It would also be great to improve this part https://github.com/elastic/kibana/pull/140516#discussion_r975020917 in the following PR

Reply:
I'll try to include it now